perun-core/src/main/java/cz/metacentrum/perun/core/entry/GroupsManagerEntry.java

```diff
@@ -896,7 +896,7 @@ public void synchronizeGroups(PerunSession sess) throws InternalErrorException,
 		return getGroupsManagerBl().getAllMemberGroups(sess, member);
 	}
 
-	public List<RichGroup> getRichGroupsAssignedToResourceWithAttributesByNames(PerunSession sess, Resource resource, List<String> attrNames) throws InternalErrorException, WrongAttributeAssignmentException, ResourceNotExistsException, PrivilegeException {
+	public List<RichGroup> getRichGroupsAssignedToResourceWithAttributesByNames(PerunSession sess, Resource resource, List<String> attrNames) throws InternalErrorException, ResourceNotExistsException, PrivilegeException, GroupResourceMismatchException {
```

Comment:
Since group is not between params you shouldn't throw ``GroupResourceMismatchException``. It should be solved once ``getRichGroupsWithAttributesAssignedToResource()`` is fixed in GroupsManagerBl as noted in my previous comments.